play-json/jvm/src/main/scala/play/api/libs/json/EnvReads.scala

```diff
@@ -528,6 +532,72 @@ trait EnvReads {
     }
   }
 
+  private def jdurationNumberReads(unit: TemporalUnit) =
+    Reads[JDuration] {
+      case JsNumber(n) if !n.ulp.isValidLong =>
+        JsError("error.invalid.longDuration")
+
+      case JsNumber(n) => JsSuccess(JDuration.of(n.toLong, unit))
+      case _ => JsError("error.expected.duration")
+    }
+
+  /**
+   * Deserializer of Java Duration from an integer (long) number,
+   * using the specified temporal unit.
+   */
+  def javaDurationNumberReads(unit: TemporalUnit): Reads[JDuration] =
+    jdurationNumberReads(unit)
+
+  /** Deserializer of Java Duration from a number of milliseconds. */
+  val javaDurationMillisReads: Reads[JDuration] =
+    javaDurationNumberReads(ChronoUnit.MILLIS)
+
+  /**
+   * Deserializer of Java Duration, from either a time-based amount of time
+   * (string representation such as '34.5 seconds'),
+   * or from a number of milliseconds (see [[javaDurationMillisReads]]).
+   */
+  implicit val DefaultJavaDurationReads: Reads[JDuration] = Reads[JDuration] {
+    case JsString(repr) => try {
+      JsSuccess(JDuration.parse(repr))
+    } catch {
+      case _: Exception => JsError("error.invalid.duration")
```

Comment:
Should be `case _: DateTimeParseException` since that's the only exception that indicates an invalid duration.